examples/sam/README.md

```diff
@@ -0,0 +1,125 @@
+- events - Invocation events that you can use to invoke the function.
+- template.yaml - A template that defines the application's AWS resources.
+
+The application uses several AWS resources, including Lambda functions and an API Gateway API. These resources are defined in the `template.yaml` file in this project. You can update the template to add AWS resources through the same deployment process that updates your application code.
+
+If you prefer to use an integrated development environment (IDE) to build and test your application, you can use the AWS Toolkit.  
+The AWS Toolkit is an open source plug-in for popular IDEs that uses the SAM CLI to build and deploy serverless applications on AWS. The AWS Toolkit also adds a simplified step-through debugging experience for Lambda function code. See the following links to get started.
+
+* [CLion](https://docs.aws.amazon.com/toolkit-for-jetbrains/latest/userguide/welcome.html)
+* [GoLand](https://docs.aws.amazon.com/toolkit-for-jetbrains/latest/userguide/welcome.html)
+* [IntelliJ](https://docs.aws.amazon.com/toolkit-for-jetbrains/latest/userguide/welcome.html)
+* [WebStorm](https://docs.aws.amazon.com/toolkit-for-jetbrains/latest/userguide/welcome.html)
+* [Rider](https://docs.aws.amazon.com/toolkit-for-jetbrains/latest/userguide/welcome.html)
+* [PhpStorm](https://docs.aws.amazon.com/toolkit-for-jetbrains/latest/userguide/welcome.html)
+* [PyCharm](https://docs.aws.amazon.com/toolkit-for-jetbrains/latest/userguide/welcome.html)
+* [RubyMine](https://docs.aws.amazon.com/toolkit-for-jetbrains/latest/userguide/welcome.html)
+* [DataGrip](https://docs.aws.amazon.com/toolkit-for-jetbrains/latest/userguide/welcome.html)
+* [VS Code](https://docs.aws.amazon.com/toolkit-for-vscode/latest/userguide/welcome.html)
+* [Visual Studio](https://docs.aws.amazon.com/toolkit-for-visual-studio/latest/user-guide/welcome.html)
+
+## Prepare the project
+
+Before deploying this example, copy the Lambda Function Code to `src/handlers`
+
+```bash
+cp -R ../lambda-functions/* ./src/handlers
+```
+
+## Deploy the sample application
+
+The Serverless Application Model Command Line Interface (SAM CLI) is an extension of the AWS CLI that adds functionality for building and testing Lambda applications. It uses Docker to run your functions in an Amazon Linux environment that matches Lambda. It can also emulate your application's build environment and API.
+
+To use the SAM CLI, you need the following tools.
+
+* SAM CLI - [Install the SAM CLI](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html)
+* Node.js - [Install Node.js 14](https://nodejs.org/en/), including the NPM package management tool.
+* Docker - [Install Docker community edition](https://hub.docker.com/search/?type=edition&offering=community)
+
+To build and deploy your application for the first time, run the following in your shell:
+
+```bash
+sam build
```

Comment:
I am not a SAM user so this was the first time using it on this machine. When running `sam build` I got this warning/notice:
```sh
❯ sam build                                                                                                                                 
Using esbuild for bundling Node.js and TypeScript is a beta feature.
Please confirm if you would like to proceed with using esbuild to build your function.
You can also enable this beta feature with 'sam build --beta-features'. [y/N]: 
```

I know we need `esbuild` because I have context on this project so I said yes, but should we say something about this for newcomers?

Reply:
I firstly ran this command while in the `examples/sam` folder (root dir of the examples) and got:
```sh
~/Codes/aws-lambda-powertools-typescript/examples/sam pr/bpauwels/674*
❯ sam build                                                                                                                              
Using esbuild for bundling Node.js and TypeScript is a beta feature.
Please confirm if you would like to proceed with using esbuild to build your function.
You can also enable this beta feature with 'sam build --beta-features'. [y/N]: y   

Experimental features are enabled for this session.
Visit the docs page to learn more about the AWS Beta terms https://aws.amazon.com/service-terms/.

Building codeuri: /Users/aamorosi/Codes/aws-lambda-powertools-typescript/examples/sam runtime: nodejs14.x metadata: {'BuildMethod': 'esbuild', 'BuildProperties': {'Minify': True, 'Target': 'es2020', 'Sourcemap': 'true,', 'EntryPoints': ['src/handlers/get-all-items.ts']}} architecture: x86_64 functions: ['getAllItemsFunction']
package.json file not found. Bundling source without dependencies.
Running NodejsNpmEsbuildBuilder:EsbuildBundle

Build Failed
Error: NodejsNpmEsbuildBuilder:EsbuildBundle - Esbuild Failed: cannot find esbuild
```

Reply:
I then went into `examples/sam/src/handlers` (which is where the only `package.json` is found and `esbuild` is mentioned as `devDependency`), and tried again after installing the dependencies:
```sh
~/Codes/aws-lambda-powertools-typescript/examples/sam pr/bpauwels/674*
❯ cd src/handlers                                                                                                                    

~/Codes/aws-lambda-powertools-typescript/examples/sam/src/handlers pr/bpauwels/674*
❯ npm i                                                                                                                                     

added 158 packages, and audited 159 packages in 19s

25 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

~/Codes/aws-lambda-powertools-typescript/examples/sam/src/handlers pr/bpauwels/674* 20s
❯ sam build                                                                                                                                 
Error: Template file not found at /Users/aamorosi/Codes/aws-lambda-powertools-typescript/examples/sam/src/handlers/template.yml
```

Reply:
I was running this on a clean clone, so I thought that maybe I had to ran `npm i` in the project directory. My thought process was that since `esbuild` is specified there as a `devDependency` then the error should go away.

That also didn't work, so I am not sure how to proceed.

Reply:
I got the sam error on the latest SAM CLI (v 1.46.0)

Reply:
Fixed with my last commit.
I was on and older beta version of sam cli that did not have this warning. I have added the --beta-features parameters to the instructions and explained it.